test/apps/wboit/app.js

```diff
@@ -1,34 +1,189 @@
+          [-122.48042337175899, 37.830085010427176],
+          [-122.47769563278436, 37.83049279439961],
+          [-122.47803148135057, 37.83189437487894],
+          [-122.4807592721423, 37.83148659090809]
+        ]
+      ]
+    }
+  },
+  {
+    type: 'Feature',
+    properties: {
+      elevation: 2500,
+      fillColor: [0, 0, 200]
+    },
+    geometry: {
+      type: 'Polygon',
+      coordinates: [
+        [
+          [-122.47796926383656, 37.809650033000324],
+          [-122.47796926383656, 37.80803536584605],
+          [-122.47501561198487, 37.80803532897342],
+          [-122.47501554739789, 37.80964999612554],
+          [-122.47796926383656, 37.809650033000324]
+        ]
+      ]
+    }
+  }
+];
+
+export default class App extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      opacity: 0.5,
+      wireframe: true,
+      lightMode: 2,
+      wboit: true
+    };
+  }
+
+  render() {
+    const {opacity, wireframe, lightMode, wboit} = this.state;
+
+    let material = new PhongMaterial();
+    let lightingEffect = new LightingEffect();
+
+    if (lightMode === 2) {
+      // Ambient Light only / Flat
+      material = new PhongMaterial({
+        ambient: 1.0,
+        diffuse: 0.0,
+        shininess: 32,
+        specularColor: [255, 255, 255]
+      });
+
+      lightingEffect = new LightingEffect({
+        light1: new AmbientLight({
+          color: [255, 255, 255],
+          intensity: 1.0
+        })
+      });
+    } else if (lightMode === 3) {
+      // Single Directional
+      material = new PhongMaterial({
+        ambient: 0.5,
+        diffuse: 0.5,
+        shininess: 32,
+        specularColor: [255, 255, 255]
+      });
+
+      lightingEffect = new LightingEffect({
+        light1: new AmbientLight({
+          color: [255, 255, 255],
+          intensity: 1.0
+        }),
+        light2: new DirectionalLight({
+          color: [255, 255, 255],
+          intensity: 1.0,
+          direction: [1, 1, 0]
+        })
+      });
+    }
+
+    const options = {
+      data,
       getPolygon: f => f.geometry.coordinates,
-      getFillColor: f => [200 + Math.random() * 55, 0, 0],
+      getFillColor: f => f.properties.fillColor,
       getLineColor: f => [0, 0, 0, 255],
-      getLineDashArray: f => [20, 0],
-      getLineWidth: f => 20,
-      getElevation: f => Math.random() * 1000,
+      getElevation: f => f.properties.elevation,
+      pickable: false,
       extruded: true,
-      opacity: 0.5,
-      pickable: false
-    })
-  ]
-});
-
-// For automated test cases
-/* global document */
-document.body.style.margin = '0px';
+      opacity,
+      wireframe,
+      material
+    };
+
+    const layers = [];
+    if (wboit) {
+      layers.push(new WBOITLayer({id: 'WBOITLayer', ...options}));
+    } else {
+      layers.push(new SolidPolygonLayer({id: 'SolidPolygonLayer', ...options}));
+    }
+
+    const mkButton = (label, name, value) => (
```

Comment:
This could be a react component as well.

Reply:
In that case, I will need to have a global state, making things more complex.